docs/reference/snapcraft-yaml.rst

```diff
@@ -3,14 +3,137 @@
 snapcraft.yaml
 ==============
 
+This page provides brief descriptions and examples for each key that can be included
+in a snap's project file.
+
 
 Top-level keys
 --------------
 
+The top-level keys and values in snapcraft.yaml provide the snap build process, and the
+store, with the overarching details of a snap.
+
+Top-level details include a snap's name, version and description, alongside operational
+values such as its confinement level and supported architectures.
+
+.. kitbash-field:: project.Project name
+
+.. kitbash-field:: craft_application.models.project.Project title
+
+.. kitbash-field:: project.Project version
+
+.. kitbash-field:: craft_application.models.project.Project license
+
+.. kitbash-field:: craft_application.models.project.Project summary
+
+.. kitbash-field:: craft_application.models.project.Project description
+
+.. kitbash-field:: project.Project adopt_info
+
+.. kitbash-field:: project.Project type
+
+.. kitbash-field:: craft_application.models.project.Project base
+
+.. kitbash-field:: project.Project build_base
+
+.. kitbash-field:: project.Project grade
+
+.. kitbash-field:: project.Project confinement
+
+.. kitbash-field:: project.Project source_code
+
+.. kitbash-field:: project.Project contact
+
+.. kitbash-field:: project.Project website
+
+.. kitbash-field:: project.Project issues
+
+.. kitbash-field:: project.Project donation
+
+.. kitbash-field:: project.Project compression
+
+.. kitbash-field:: project.Project icon
+
+.. kitbash-field:: project.Project layout
+
+.. kitbash-field:: project.Project passthrough
+
+.. kitbash-field:: project.Project assumes
+
+.. kitbash-field:: project.Project slots
+
+.. kitbash-field:: project.Project lint
+
+.. kitbash-field:: project.Project epoch
+
+.. kitbash-field:: project.Project system_usernames
+
+.. kitbash-field:: project.Project environment
+
+.. kitbash-field:: project.Project build_packages
+
+.. kitbash-field:: project.Project build_snaps
+
+.. kitbash-field:: project.Project ua_services
+
+.. kitbash-field:: project.Project provenance
+
+.. kitbash-field:: project.Project platforms
+
+.. kitbash-field:: project.Project architectures
+
+.. kitbash-field:: project.Project apps
+
+.. kitbash-field:: craft_application.models.project.Project parts
+
+.. kitbash-field:: project.Project hooks
```

Comment:
These should be similar to the apps, like `hooks.<hook-type>.passthrough`.

side note: `<hook-type>` may be more precise than `<hook-name>` since there is a finite set of hooks.  I don't think the list of hook types is in our project model, so adding the types is out of scope.